pkg/csi_driver/multishare_ops_manager.go

```diff
@@ -84,13 +84,20 @@ func (m *MultishareOpsManager) setupEligibleInstanceAndStartWorkflow(ctx context
 	}
 
 	// Check if share already part of an existing instance.
-	shares, err := m.cloud.File.ListShares(ctx, &file.ListFilter{Project: m.cloud.Project, Location: "-", InstanceName: "-"})
+	regions, err := m.listRegions(req.GetAccessibilityRequirements())
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	for _, s := range shares {
-		if s.Name == shareName {
-			return nil, s, nil
+	for _, region := range regions {
+		shares, err := m.cloud.File.ListShares(ctx, &file.ListFilter{Project: m.cloud.Project, Location: region, InstanceName: "-"})
```

Comment:
we should reuse the shares list in the runEligibleInstanceCheck()

Reply:
it should be in a seperate pr since it's a much bigger change and it's only incremental optimization